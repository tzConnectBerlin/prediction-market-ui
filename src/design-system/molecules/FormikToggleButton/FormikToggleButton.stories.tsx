import { Story, Meta } from '@storybook/react/types-6-0';
import * as Yup from 'yup';
import { Field, Formik } from 'formik';
import { FormikToggleButton, FormikToggleButtonProps } from './FormikToggleButton';

export default {
  title: 'Molecule/FormikToggleButton',
  component: FormikToggleButton,
} as Meta;

const OutComePrice = Yup.object().shape({
  outCome: Yup.string().required('Required'),
});

const items = [
  {
    value: 'Yes',
    label: '12$',
  },
  {
    value: 'No',
    label: '8$',
  },
];

const formikProps = {
  initialValues: {
    outCome: '',
  },
  validationSchema: OutComePrice,
  onSubmit: () => {},
};

const Template: Story<FormikToggleButtonProps> = (args) => (
  <Formik {...formikProps}>
    <Field component={FormikToggleButton} {...args} />
  </Formik>
);

export const Default = Template.bind({});
Default.args = {
  label: 'Outcome',
  required: true,
  onChange: (val: any) => console.log(val),
  toggleButtonItems: items,
  value: items[0].value,
};

export const Disabled = Template.bind({});
Disabled.args = {
  label: 'Outcome',
  required: true,
  onChange: (val: any) => console.log(val),
  toggleButtonItems: items,
  value: items[0].value,
  disabled: true,
};
