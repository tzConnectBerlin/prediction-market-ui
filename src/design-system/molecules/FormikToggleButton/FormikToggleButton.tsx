import * as React from 'react';
import styled from '@emotion/styled';
import {
  FormControl,
  PaletteOptions,
  Theme,
  ToggleButton,
  ToggleButtonGroup,
  ToggleButtonGroupProps,
  useTheme,
} from '@mui/material';
import { FieldProps } from 'formik';
import { CustomInputChipProps, CustomInputLabel } from '../CustomInputLabel';

type PaletteOptionsType = keyof PaletteOptions;

interface StyledToggleButtonProps {
  theme: Theme;
  color: PaletteOptionsType;
}

const StyledToggleButtonGroup = styled(ToggleButtonGroup)<{ theme: Theme }>`
  margin-top: ${({ theme }) => theme.spacing(1)};
  margin-bottom: ${({ theme }) => theme.spacing(1)};
  button.MuiToggleButton-root {
    border-color: transparent;
    border-radius: 0;
    background-color: ${({ theme }) => theme.palette.grey[300]};
    padding: ${({ theme }) => theme.spacing(1 / 3)};
    &.Mui-disabled,
    &[disabled] {
      opacity: 0.8;
    }
  }
`;

const StyledToggleButton = styled(ToggleButton)<StyledToggleButtonProps>`
  text-transform: capitalize;
  &.Mui-selected {
    color: ${({ theme, color }) => theme.palette[color].main};
    font-weight: bold;

    &.MuiToggleButtonGroup-grouped:first-of-type {
      background-color: ${({ theme, color }) => theme.palette[color].dark};
      border-left-color: ${({ theme, color }) => theme.palette[color].main};
    }
    &.MuiToggleButtonGroup-grouped:last-of-type {
      border-right-color: ${({ theme, color }) => theme.palette[color].main};
      background-color: ${({ theme, color }) => theme.palette[color].dark};
    }
  }
`;

interface InternalToggleButtonProps extends FieldProps {
  toggleButtonItems: ToggleButtonItems[];
  tooltip?: boolean;
  tooltipText?: string;
  helpMessage?: string;
  label?: string;
  required?: boolean;
  disabled?: boolean;
}

export interface ToggleButtonItems {
  value: string;
  label: string;
  selectedColor?: PaletteOptionsType | any;
}

export type FormikToggleButtonProps = InternalToggleButtonProps &
  ToggleButtonGroupProps &
  CustomInputChipProps;

const defaultChip = false;
const defaultDisabled = false;

export const FormikToggleButton: React.FC<FormikToggleButtonProps> = ({
  toggleButtonItems,
  field: { name, value },
  form: { setFieldValue },
  label,
  required,
  helpMessage,
  tooltip,
  tooltipText,
  chip = defaultChip,
  chipText,
  chipIcon,
  chipOnClick,
  disabled = defaultDisabled,
  children,
  onChange,
  ...rest
}) => {
  const theme = useTheme();
  const handleAlignment = (event: React.MouseEvent<HTMLElement>, newAlignment: string | null) => {
    if (newAlignment !== null) {
      onChange && onChange(event, newAlignment);
      setFieldValue(name, newAlignment);
    }
  };

  return (
    <FormControl>
      <CustomInputLabel
        shrink
        htmlFor={name}
        label={label}
        required={required}
        disabled={disabled}
        helpMessage={helpMessage}
        tooltipText={tooltipText}
        tooltip={tooltip}
        chip={chip}
        chipText={chipText}
        chipIcon={chipIcon}
        chipOnClick={chipOnClick}
      />
      <StyledToggleButtonGroup
        {...rest}
        exclusive
        fullWidth
        value={value}
        onChange={handleAlignment}
        id={name}
        theme={theme}
      >
        {toggleButtonItems.map((item: ToggleButtonItems, index) => (
          <StyledToggleButton
            value={item.label}
            key={`${item.value}-${index}`}
            disabled={disabled}
            color={item?.selectedColor ?? 'success'}
            theme={theme}
          >
            {item.label} ({item.value})
          </StyledToggleButton>
        ))}
      </StyledToggleButtonGroup>
    </FormControl>
  );
};
