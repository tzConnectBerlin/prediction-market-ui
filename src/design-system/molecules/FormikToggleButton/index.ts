export type { FormikToggleButtonProps } from './FormikToggleButton';
export { FormikToggleButton, FormikToggleButton as default } from './FormikToggleButton';
