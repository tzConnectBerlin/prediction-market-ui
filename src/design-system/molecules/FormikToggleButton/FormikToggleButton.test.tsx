import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent, act } from '@testing-library/react';
import * as Yup from 'yup';
import { FastField, Form, Formik } from 'formik';
import { FormikToggleButton } from './FormikToggleButton';

const OutComePrice = Yup.object().shape({
  outCome: Yup.string().required('Required'),
});

const items = [
  {
    value: '12$',
    label: 'Yes',
  },
  {
    value: '8$',
    label: 'No',
  },
];

const formikProps = {
  initialValues: {
    outCome: '',
  },
  validationSchema: OutComePrice,
  onSubmit: () => {},
};

const baseArgs = {
  label: 'Outcome',
  required: true,
  toggleButtonItems: items,
  value: items[0].value,
};

const defaultArgs = {
  ...baseArgs,
  onChange: jest.fn(),
};

const WrappedComponent: React.FC<any> = (props) => (
  <Formik {...formikProps}>
    <Form>
      <FastField component={FormikToggleButton} {...props} />
    </Form>
  </Formik>
);

describe('Snapshot - render FormikToggleButton', () => {
  it('renders correctly with default props', () => {
    const ToggleButton = renderer.create(<WrappedComponent {...defaultArgs} />).toJSON();
    expect(ToggleButton).toMatchSnapshot();
  });

  it('renders correctly in disabled mode', () => {
    const ToggleButton = renderer.create(<WrappedComponent {...defaultArgs} disabled />).toJSON();
    expect(ToggleButton).toMatchSnapshot();
  });
});

describe('Element testing FormikToggleButton Component', () => {
  it('render correctly FormikToggleButton with default props', async () => {
    const { getByText } = render(<WrappedComponent {...defaultArgs} />);

    expect(getByText(/Yes/i)).toBeInTheDocument();
    expect(getByText(/No/i)).toBeInTheDocument();
  });

  it('check onChange gets called', async () => {
    const { findByRole } = render(<WrappedComponent {...defaultArgs} />);
    const component = await findByRole('group');
    // const component = await findByRole('button', { pressed: true });
    act(() => {
      fireEvent.click(component.children[0], { target: { value: 'yes' } });
    });
    expect(defaultArgs.onChange).toBeCalled();
  });
});
