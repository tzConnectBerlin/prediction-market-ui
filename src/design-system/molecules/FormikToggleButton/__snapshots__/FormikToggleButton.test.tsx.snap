// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot - render FormikToggleButton renders correctly in disabled mode 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  min-width: 0;
  padding: 0;
  margin: 0;
  border: 0;
  vertical-align: top;
}

.emotion-1 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-basis: 91.666667%;
  -ms-flex-preferred-size: 91.666667%;
  flex-basis: 91.666667%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  max-width: 91.666667%;
}

.emotion-3 {
  color: rgba(0, 0, 0, 0.6);
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.4375em;
  letter-spacing: 0.00938em;
  padding: 0;
  position: relative;
  display: block;
  transform-origin: top left;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  max-width: 133%;
  position: absolute;
  left: 0;
  top: 0;
  -webkit-transform: translate(0, -1.5px) scale(0.75);
  -moz-transform: translate(0, -1.5px) scale(0.75);
  -ms-transform: translate(0, -1.5px) scale(0.75);
  transform: translate(0, -1.5px) scale(0.75);
  -webkit-transition: color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,-webkit-transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,max-width 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,max-width 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
}

.emotion-3.Mui-focused {
  color: #1976d2;
}

.emotion-3.Mui-disabled {
  color: rgba(0, 0, 0, 0.38);
}

.emotion-3.Mui-error {
  color: #d32f2f;
}

.emotion-4 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.75;
  letter-spacing: 0.00938em;
}

.emotion-4.truncate {
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 3;
}

.emotion-5.Mui-error {
  color: #d32f2f;
}

.emotion-6 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  border-radius: 4px;
  width: 100%;
  margin-top: 8px;
  margin-bottom: 8px;
}

.emotion-6 .MuiToggleButtonGroup-grouped:not(:first-of-type) {
  margin-left: -1px;
  border-left: 1px solid transparent;
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;
}

.emotion-6 .MuiToggleButtonGroup-grouped:not(:last-of-type) {
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;
}

.emotion-6 .MuiToggleButtonGroup-grouped.Mui-selected+.MuiToggleButtonGroup-grouped.Mui-selected {
  border-left: 0;
  margin-left: 0;
}

.emotion-6 button.MuiToggleButton-root {
  border-color: transparent;
  border-radius: 0;
  background-color: #e0e0e0;
  padding: 2.6666666666666665px;
}

.emotion-6 button.MuiToggleButton-root.Mui-disabled,
.emotion-6 button.MuiToggleButton-root[disabled] {
  opacity: 0.8;
}

.emotion-7 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  letter-spacing: 0.02857em;
  text-transform: uppercase;
  border-radius: 4px;
  padding: 11px;
  border: 1px solid rgba(0, 0, 0, 0.12);
  color: rgba(0, 0, 0, 0.54);
  width: 100%;
}

.emotion-7::-moz-focus-inner {
  border-style: none;
}

.emotion-7.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-7 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-7.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
  border: 1px solid rgba(0, 0, 0, 0.12);
}

.emotion-7:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(0, 0, 0, 0.04);
}

@media (hover: none) {
  .emotion-7:hover {
    background-color: transparent;
  }
}

.emotion-7.Mui-selected {
  color: #2e7d32;
  background-color: rgba(46, 125, 50, 0.08);
}

.emotion-7.Mui-selected:hover {
  background-color: rgba(46, 125, 50, 0.12);
}

@media (hover: none) {
  .emotion-7.Mui-selected:hover {
    background-color: rgba(46, 125, 50, 0.08);
  }
}

.emotion-7.Mui-selected {
  color: #2e7d32;
}

.emotion-7.Mui-selected.MuiToggleButtonGroup-grouped:first-of-type {
  background-color: #1b5e20;
  border-left-color: #2e7d32;
}

.emotion-7.Mui-selected.MuiToggleButtonGroup-grouped:last-of-type {
  border-right-color: #2e7d32;
  background-color: #1b5e20;
}

<form
  action="#"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="MuiFormControl-root emotion-0"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-undefined emotion-1"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-11 emotion-2"
      >
        <label
          className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary Mui-disabled Mui-required emotion-3"
          data-shrink={true}
        >
          <span
            className="MuiTypography-root MuiTypography-subtitle1  emotion-4"
            lines={3}
          >
            Outcome
          </span>
          <span
            aria-hidden={true}
            className="label-asterisk MuiFormLabel-asterisk emotion-5"
          >
            â€‰
            *
          </span>
        </label>
      </div>
    </div>
    <div
      className="MuiToggleButtonGroup-root MuiToggleButtonGroup-fullWidth emotion-6"
      role="group"
    >
      <button
        aria-pressed={false}
        className="MuiButtonBase-root Mui-disabled MuiToggleButton-root Mui-disabled MuiToggleButton-fullWidth MuiToggleButton-sizeMedium MuiToggleButton-success MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal emotion-7"
        disabled={true}
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={-1}
        type="button"
        value="Yes (12$)"
      >
        Yes (12$)
         (
        Yes
        )
      </button>
      <button
        aria-pressed={false}
        className="MuiButtonBase-root Mui-disabled MuiToggleButton-root Mui-disabled MuiToggleButton-fullWidth MuiToggleButton-sizeMedium MuiToggleButton-success MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal emotion-7"
        disabled={true}
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={-1}
        type="button"
        value="No (8$)"
      >
        No (8$)
         (
        No
        )
      </button>
    </div>
  </div>
</form>
`;

exports[`Snapshot - render FormikToggleButton renders correctly with default props 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  min-width: 0;
  padding: 0;
  margin: 0;
  border: 0;
  vertical-align: top;
}

.emotion-1 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-basis: 91.666667%;
  -ms-flex-preferred-size: 91.666667%;
  flex-basis: 91.666667%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  max-width: 91.666667%;
}

.emotion-3 {
  color: rgba(0, 0, 0, 0.6);
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.4375em;
  letter-spacing: 0.00938em;
  padding: 0;
  position: relative;
  display: block;
  transform-origin: top left;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  max-width: 133%;
  position: absolute;
  left: 0;
  top: 0;
  -webkit-transform: translate(0, -1.5px) scale(0.75);
  -moz-transform: translate(0, -1.5px) scale(0.75);
  -ms-transform: translate(0, -1.5px) scale(0.75);
  transform: translate(0, -1.5px) scale(0.75);
  -webkit-transition: color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,-webkit-transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,max-width 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,max-width 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
}

.emotion-3.Mui-focused {
  color: #1976d2;
}

.emotion-3.Mui-disabled {
  color: rgba(0, 0, 0, 0.38);
}

.emotion-3.Mui-error {
  color: #d32f2f;
}

.emotion-4 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.75;
  letter-spacing: 0.00938em;
}

.emotion-4.truncate {
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 3;
}

.emotion-5.Mui-error {
  color: #d32f2f;
}

.emotion-6 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  border-radius: 4px;
  width: 100%;
  margin-top: 8px;
  margin-bottom: 8px;
}

.emotion-6 .MuiToggleButtonGroup-grouped:not(:first-of-type) {
  margin-left: -1px;
  border-left: 1px solid transparent;
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;
}

.emotion-6 .MuiToggleButtonGroup-grouped:not(:last-of-type) {
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;
}

.emotion-6 .MuiToggleButtonGroup-grouped.Mui-selected+.MuiToggleButtonGroup-grouped.Mui-selected {
  border-left: 0;
  margin-left: 0;
}

.emotion-6 button.MuiToggleButton-root {
  border-color: transparent;
  border-radius: 0;
  background-color: #e0e0e0;
  padding: 2.6666666666666665px;
}

.emotion-6 button.MuiToggleButton-root.Mui-disabled,
.emotion-6 button.MuiToggleButton-root[disabled] {
  opacity: 0.8;
}

.emotion-7 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  letter-spacing: 0.02857em;
  text-transform: uppercase;
  border-radius: 4px;
  padding: 11px;
  border: 1px solid rgba(0, 0, 0, 0.12);
  color: rgba(0, 0, 0, 0.54);
  width: 100%;
}

.emotion-7::-moz-focus-inner {
  border-style: none;
}

.emotion-7.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-7 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-7.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
  border: 1px solid rgba(0, 0, 0, 0.12);
}

.emotion-7:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(0, 0, 0, 0.04);
}

@media (hover: none) {
  .emotion-7:hover {
    background-color: transparent;
  }
}

.emotion-7.Mui-selected {
  color: #2e7d32;
  background-color: rgba(46, 125, 50, 0.08);
}

.emotion-7.Mui-selected:hover {
  background-color: rgba(46, 125, 50, 0.12);
}

@media (hover: none) {
  .emotion-7.Mui-selected:hover {
    background-color: rgba(46, 125, 50, 0.08);
  }
}

.emotion-7.Mui-selected {
  color: #2e7d32;
}

.emotion-7.Mui-selected.MuiToggleButtonGroup-grouped:first-of-type {
  background-color: #1b5e20;
  border-left-color: #2e7d32;
}

.emotion-7.Mui-selected.MuiToggleButtonGroup-grouped:last-of-type {
  border-right-color: #2e7d32;
  background-color: #1b5e20;
}

<form
  action="#"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="MuiFormControl-root emotion-0"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-undefined emotion-1"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-11 emotion-2"
      >
        <label
          className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required emotion-3"
          data-shrink={true}
        >
          <span
            className="MuiTypography-root MuiTypography-subtitle1  emotion-4"
            lines={3}
          >
            Outcome
          </span>
          <span
            aria-hidden={true}
            className="label-asterisk MuiFormLabel-asterisk emotion-5"
          >
            â€‰
            *
          </span>
        </label>
      </div>
    </div>
    <div
      className="MuiToggleButtonGroup-root MuiToggleButtonGroup-fullWidth emotion-6"
      role="group"
    >
      <button
        aria-pressed={false}
        className="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-fullWidth MuiToggleButton-sizeMedium MuiToggleButton-success MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal emotion-7"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
        value="Yes (12$)"
      >
        Yes (12$)
         (
        Yes
        )
      </button>
      <button
        aria-pressed={false}
        className="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-fullWidth MuiToggleButton-sizeMedium MuiToggleButton-success MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal emotion-7"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
        value="No (8$)"
      >
        No (8$)
         (
        No
        )
      </button>
    </div>
  </div>
</form>
`;
